client/app/dbaas/logs/detail/inputs/home/logs-inputs-home.controller.js

```diff
@@ -0,0 +1,194 @@
+        this.inputReload = this.ControllerHelper.request.getHashLoader({
+            loaderFunction: () => this.LogsInputsService.getInputDetail(this.serviceName, inputId)
+        });
+
+        return this.inputReload.load()
+            .then(input => {
+                this.inputs.data.forEach((inputItem, inputIndex) => {
+                    if (inputItem.info.inputId === input.info.inputId) {
+                        this.inputs.data[inputIndex] = input;
+                    }
+                });
+                return input;
+            });
+    }
+
+    /**
+     * Runs all the loaders to fetch data from APIs
+     *
+     * @memberof LogsInputsCtrl
+     */
+    _runLoaders () {
+        this.inputs.load();
+        this.quota.load();
+    }
+
+    /**
+     * Sets the state of the input to Processing
+     *
+     * @param {any} input
+     * @memberof LogsInputsCtrl
+     */
+    _setInputToProcessing (input) {
+        input.info.status = this.LogsInputsConstant.status.PROCESSING;
+        this.LogsInputsService.transformInput(input);
+    }
+
+    /**
+     * Opens the info pop-up for the input
+     *
+     * @param {any} input - the input for which info is to be displayed
+     * @memberof LogsInputsCtrl
+     */
+    info (input) {
+        this.CloudMessage.flushChildMessage();
+        this.ControllerHelper.modal.showModal({
+            modalConfig: {
+                templateUrl: "app/dbaas/logs/detail/inputs/home/info/logs-inputs-home-info.html",
+                controller: "LogsInputsHomeInfoModalCtrl",
+                controllerAs: "ctrl",
+                resolve: {
+                    currentInput: () => input
+                }
+            }
+        });
+    }
+
+    /**
+     * Shows the confirmation modal box for input deletion confirmation
+     * and deletes the input if the user confirms the deletion
+     *
+     * @param {any} input - the input object
+     * @memberof LogsInputsCtrl
+     */
+    showDeleteConfirm (input) {
+        this.CloudMessage.flushChildMessage();
+        return this.ControllerHelper.modal.showDeleteModal({
+            titleText: this.$translate.instant("inputs_delete"),
+            text: this.$translate.instant("inputs_delete_message", { input: input.info.title })
+        }).then(() => this._delete(input));
+    }
+
+    /**
+     * Restarts the input
+     *
+     * @param {any} input - the input to be restarted
+     * @memberof LogsInputsCtrl
+     */
+    restartInput (input) {
+        this.CloudMessage.flushChildMessage();
+        this.CloudMessage.info(this.$translate.instant("inputs_restarting", { inputTitle: input.info.title }));
+        this._setInputToProcessing(input);
+        this.processInput = this.ControllerHelper.request.getHashLoader({
+            loaderFunction: () => this.LogsInputsService.restartInput(this.serviceName, input)
+        });
+        this.processInput.load()
+            .then(() => this._reloadInputDetail(input.info.inputId),
```

Comment:
remove that if you make the change above

Reply:
This is done as above